File: ksd-vue-admin/public/src/api/sysroles.js
import request from '@/request/index.js'

/**
 * 查询用户角色信息
 */
export const FindData = () => {
  return request.post(`/sys/role/list`, {})
}

/**
 * 查询系统角色列表并分页
 */
export const LoadData = (data) => {
  return request.post(`/sys/role/load`, data)
}

/**
 * 根据id查询系统角色信息
 */
export const GetById = (id) => {
  return request.post(`/sys/role/get/${id}`)
}

/**
 * 保存系统角色
 */
export const SaveData = (data) => {
  return request.post(`/sys/role/save`, data)
}

/**
 * 更新系统角色信息
 */
export const UpdateData = (data) => {
  return request.post(`/sys/role/update`, data)
}

/**
 * 根据id删除系统角色信息
 */
export const DelById = (id) => {
  return request.post(`/sys/role/del/${id}`)
}

/**
 * 根据ids批量删除系统角色信息
 */
export const DelByIds = (ids) => {
  return request.post(`/sys/role/dels?ids=${ids}`)
}

/**
 * 系统角色启用和未启用
 */
export const UpdateStatusSysRole = (data) => {
  return request.post(`/sys/role/update/status`, data)
}

/**
 * 查询角色对应的菜单
 */
export const SelectRoleMenus = (roleId) => {
  return request.post(`/sys/role/menu/list?roleId=${roleId}`, {})
}

/**
 * 保存角色对应的菜单
 */
export const SaveRoleMenus = (data = {}) => {
  return request.post(`/sys/role/menu/save`, data)
}

/**
 * 查询角色对应的apis
 */
export const SelectRoleApis = (roleId) => {
  return request.post(`/sys/role/api/list?roleId=${roleId}`, {})
}

/**
 * 保存角色对应的apis
 */
export const SaveRoleApis = (data = {}) => {
  return request.post(`/sys/role/api/save`, data)
}

/**
 * 改变角色
 */
export const ChangeRoleIdMenus = (data) => {
  return request.post(`/sys/role/api/change?roleId=${data.roleId}`, data)
}

File: xkginweb/api/router/sys/sys_roles.go
package sys

import (
	"github.com/gin-gonic/gin"
	v1 "xkginweb/api/v1"
)

// 登录路由
type SysRolesRouter struct{}

func (r *SysRolesRouter) InitSysRoleRouter(Router *gin.RouterGroup) {
	sysRolesApi := v1.WebApiGroupApp.Sys.SysRolesApi
	// 用组定义--（推荐）
	router := Router.Group("/sys")
	{
		// 查询全部角色
		router.POST("/role/list", sysRolesApi.FindRoles)
		// 保存
		router.POST("/role/save", sysRolesApi.SaveData)
		// 修改
		router.POST("/role/update", sysRolesApi.UpdateById)
		// 启用和未启用 （控制启用，发布，删除）
		router.POST("/role/update/status", sysRolesApi.UpdateStatus)
		// 删除单个 :id 获取参数的时候id := c.Param("id")，传递的时候/sys/user/del/100
		router.POST("/role/del/:id", sysRolesApi.DeleteById)
		// 删除多个  获取参数的时候ids := c.Query("ids")，传递的时候/sys/user/dels?ids=1,2,3,4
		router.POST("/role/dels", sysRolesApi.DeleteByIds)
		// 查询明细 /user/get/1/xxx
		router.POST("/role/get/:id", sysRolesApi.GetById)
		// 查询分页,搜索
		router.POST("/role/load", sysRolesApi.LoadSysRolesPage)
	}
}
